submissions/abdelmajidhafidi/main.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/*
+ * Return each line using generator
+ * @param string $path
+ */
+function getLines(string $path) {
+    $handle = fopen($path, "r");
+
+    while(!feof($handle)) {
+        yield trim(fgets($handle));
+    }
+
+    fclose($handle);
+}
+
+/**
+ * Process of getting infos from each line
+ * @param $item
+ * @param array $cities
+ */
+function getElement($item, array &$cities)
+{
+    $c = explode(",", $item);
+    if (!isset($cities[$c[0]])) {
+        $cities[$c[0]] = array();
+        $cities[$c[0]]["products"] = array();
+        $cities[$c[0]]["total"] = 0.00;
+    }
+    $cities[$c[0]]["city"] = $c[0];
+    $cities[$c[0]]["products"][] = [
+        "productName" => $c[1],
+        "productPrice" => (float)$c[2]
+    ];
+    $cities[$c[0]]["total"] += (float)$c[2];
+}
+
+$cities = array();
+
+//loop for getting elements from each lien of the input.txt file
+foreach (getLines("input_petit.txt") as $item) {
```

Comment:
I think the input file name is wrong here

Reply:
yes sorry, I correct it